js/songs/new.controller.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+(function(){
+  angular
+  .module('songs')
+  .controller('NewSongController', [
+    'SongsKitchen',
+    "$state",
+    "$stateParams",
+    NewSongControllerFunction
+  ]);
+
+  function NewSongControllerFunction(SongsKitchen,$state, $stateParams){
+    var newSongVM = this;
+    newSongVM.song = new SongsKitchen();
+    newSongVM.song.create = function(){
+      newSongVM.song.$save(newSongVM.song, function(song){
```

Comment:
Great job here with the create method and using a callback function to call `$state.go`